proof/refine/ARM/Schedule_R.thy

```diff
@@ -3040,5 +2990,662 @@ lemma ct_released_cross_weak:
                    split: kernel_object.splits)
   done
 
+lemma setReprogramTimer_corres[corres]:
+  "corres dc \<top> \<top> (modify (reprogram_timer_update (\<lambda>_. b))) (setReprogramTimer b)"
+  apply (unfold setReprogramTimer_def)
+  apply (rule corres_modify)
+  apply (simp add: state_relation_def swp_def)
+  done
+
+lemma getCurTime_corres[corres]:
+  "corres (=) \<top> \<top> (gets cur_time) (getCurTime)"
+  apply (clarsimp simp: getCurTime_def state_relation_def)
+  done
+
+lemma hd_wrap_slice:
+  "\<lbrakk>0 < count; mx \<le> length list;  start < mx\<rbrakk> \<Longrightarrow> hd (wrap_slice start count mx list) = list ! start"
+  by (auto simp: wrap_slice_def hd_drop_conv_nth)
```

Comment:
Can we move these `wrap_slice` and `refill_map` lemmas to StateRelation.thy, near where they are defined?

Reply:
Yep, moved, along with `length_wrap_slice`